src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -312,7 +312,7 @@ internal async ValueTask<SemanticModel> GetRequiredNullableDisabledSemanticModel
     /// </returns>
     public Task<SemanticModel?> GetSemanticModelAsync(CancellationToken cancellationToken = default)
     {
-        return GetSemanticModelHelperAsync(disableNullableAnalysis: false, cancellationToken);
+        return await GetSemanticModelHelperAsync(disableNullableAnalysis: false, cancellationToken);
```

Comment:
this is not a correct fix.  this is a non-async method properly calling into a Task-returning method and returning that task itself.